models/classes/runner/synchronisation/action/Pause.php

```diff
@@ -65,4 +67,16 @@ public function process()
 
         return $response;
     }
+
+    private function shouldTimerStopOnPause(): bool
+    {
+        $isTerminated = $this->getRunnerService()->isTerminated($this->getServiceContext());
```

Comment:
too many conditions and it is not so easy to understand when it should be stoped. Maybe it has sense to add extra method?
```
func shouldTimerStopOnPause() :
 return $this->shouldStopByClientsTimer
```

Reply:
I'm not sure `shouldStopByClientsTimer()` will be useful here. Especially taking into account that it is a temporary fix until proper refactoring by test runner team. We are checking just two conditions, if session was not terminated and timer target is "client". I can refactor code a bit to make it more readable.